CRM/Contact/BAO/Query.php

```diff
@@ -514,7 +514,7 @@ public function initialize($apiEntity = NULL) {
       CRM_Financial_BAO_FinancialType::buildPermissionedClause($this->_whereClause, $component);
     }
 
-    $this->_fromClause = self::fromClause($this->_tables, NULL, NULL, $this->_primaryLocation, $this->_mode);
+    $this->_fromClause = self::fromClause($this->_tables, NULL, NULL, $this->_primaryLocation, $this->_mode, $apiEntity);
```

Comment:
Just for the record, I still don't understand how that isn't just duplicating mode in a different format (you don't need to respond to this - it has been merged & it is not a 'live' concern)